internal/controlplane/handlers_repositories.go

```diff
@@ -445,3 +450,31 @@ func (s *Server) ListRemoteRepositoriesFromProvider(
 
 	return out, nil
 }
+
+func (s *Server) deleteRepositoryAndWebhook(
+	ctx context.Context,
+	repo db.Repository,
+	projectID uuid.UUID,
+	provider db.Provider,
+) error {
+	tx, err := s.store.BeginTransaction()
+	if err != nil {
+		return status.Errorf(codes.Internal, "error deleting repository")
+	}
+	defer s.store.Rollback(tx)
+
+	qtx := s.store.GetQuerierWithTransaction(tx)
+	if err := qtx.DeleteRepository(ctx, repo.ID); err != nil {
+		return status.Errorf(codes.Internal, "error deleting repository: %v", err)
+	}
+
+	if err := s.deleteWebhookFromRepository(ctx, provider, projectID, repo); err != nil {
+		return err
+	}
```

Comment:
While it doesn't particularly matter since you're using a transaction, I'd tend to put the call to remove the webhook _before_ the database delete, such that failing to clean up the webhook doesn't even need to touch the database.  (Since you're in a transaction, you won't generate a conflict between other updates to the repository and this delete until the `Commit`, at which point you will have either deleted the webhook or not.)

In general, we probably need a bit of anti-entropy here (e.g. periodically re-check that the webhook is installed, and on unexpected webhooks, try to clean them up), but I'm not going to ask you to solve that.

Reply:
>  I'd tend to put the call to remove the webhook before the database delete, such that failing to clean up the webhook doesn't even need to touch the database. 

1. User decides to delete the repository
2. User invokes the command, but it fails while deleting the repo record (DB Call), but the webhook was deleted (GH API Call)
3. User thinks this is a signal from god and decides to keep the repo
4. Webhook is now gone. Yikes!

> In general, we probably need a bit of anti-entropy here (e.g. periodically re-check that the webhook is installed, and on unexpected webhooks, try to clean them up),

Yeah, we need some sort of reconciliation mechanism to stay up to date with the state of the webhooks.

> but I'm not going to ask you to solve that.

I would like to solve this, can we discuss the exact scope in a new issue and some other details?